app/main/api/handlers/encryptWallet.ts

```diff
@@ -38,13 +37,9 @@ export default async function encryptWallet(system: AppStateS & WalletStorageS,
   Promise<JsonSerializable> {
 
   return parseParams(params)
-    .then(async (p) => Promise.all([
-      Promise.resolve(p)
-        .then(inject(newWalletStorage, system.storageFactory)),
-      Promise.resolve(p)
-        .then(inject(getUnconsolidatedData, system.appStateManager))
-        .then(inject(newWallet, system.appStateManager))
-    ]))
+    .then(inject(getUnconsolidatedData, system.appStateManager))
+    .then(newWallet)
+    .then(inject(newWalletStorage, system.storageFactory))
```

Comment:
I think this logic misses the part where the plain storage is updated with the WalletInfo (id and caption) of the newly created wallet. I believe this is the point where it should be done because it is not until this point that the id is generated, we don't have it before.

The behaviour shown is that the encrypted wallet is stored in a new folder in the filesystem but when restarting Sheikah app, no wallet is detected because the plain storage is empty.

I'll create a separate issue to refer to this.

Reply:
I've created issue #349 to talk about this. I think we should talk to the rest of the team to raise its priority if everything goes right with this PR and the other one we have pending.